process/process_windows.go

```diff
@@ -431,3 +445,132 @@ func getProcessMemoryInfo(h syscall.Handle, mem *PROCESS_MEMORY_COUNTERS) (err e
 	}
 	return
 }
+
+func getProcessHandleCount(h syscall.Handle, count *uint32) (err error) {
+	r1, _, e1 := procGetProcessHandleCount.Call(uintptr(h), uintptr(unsafe.Pointer(count)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+
+func getProcessIoCounters(h syscall.Handle, counters *IO_COUNTERS) (err error){
+	r1, _, e1 := procGetProcessIoCounters.Call(uintptr(h), uintptr(unsafe.Pointer(counters)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+func AllProcesses()(map[int32]*FilledProcess, error) {
+	allProcsSnap := w32.CreateToolhelp32Snapshot(w32.TH32CS_SNAPPROCESS, 0)
+	if allProcsSnap == 0 {
+		return nil, syscall.GetLastError()
+	}
+	procs := make(map[int32]*FilledProcess)
+
+	defer w32.CloseHandle(allProcsSnap)
+	var pe32 w32.PROCESSENTRY32
+	pe32.DwSize = uint32(unsafe.Sizeof(pe32))
+
+	for success := w32.Process32First(allProcsSnap, &pe32) ; success ; success = w32.Process32Next(allProcsSnap, &pe32) {
+		pid := pe32.Th32ProcessID
+		ppid := pe32.Th32ParentProcessID
+
+		// 0x1000 is PROCESS_QUERY_LIMITED_INFORMATION, but that constant isn't
+		// defined in syscall
+		procHandle, err := syscall.OpenProcess(0x1000, false, uint32(pid))
+		if err != nil {
+			continue
+		}
+		defer syscall.CloseHandle(procHandle)
+
+		var CPU syscall.Rusage
+		if err = syscall.GetProcessTimes(procHandle, &CPU.CreationTime, &CPU.ExitTime, &CPU.KernelTime, &CPU.UserTime); err != nil {
+			continue
+		}
+
+		var handleCount uint32
+		if err = getProcessHandleCount(procHandle, &handleCount); err != nil {
+			continue;
+		}
+
+		var pmemcounter PROCESS_MEMORY_COUNTERS
+		if err = getProcessMemoryInfo(procHandle, &pmemcounter); err != nil {
+			continue
+		}
+
+		// shell out to getprocessiocounters for io stats
+		var ioCounters IO_COUNTERS
+		if err = getProcessIoCounters(procHandle, &ioCounters); err != nil {
+			continue
+		}
+		ctime := CPU.CreationTime.Nanoseconds() / 1000000
+		
+		exename := strings.Split(convert_windows_string(pe32.SzExeFile[:]), " ")
+		utime := float64((int64(CPU.UserTime.HighDateTime) << 32) | int64(CPU.UserTime.LowDateTime)) 
+		stime := float64((int64(CPU.KernelTime.HighDateTime) << 32) | int64(CPU.KernelTime.LowDateTime))
+		username, err := get_username_for_process(procHandle)
+		procs[int32(pid)] = &FilledProcess {
+			Pid:			int32(pid),
+			Ppid:			int32(ppid),
+			Cmdline:        exename,
+			CpuTime:		cpu.TimesStat{
+				User: utime,
+				System: stime,
+				Timestamp: time.Now().UnixNano(),
+			},
+			
+			CreateTime: ctime,
+			OpenFdCount: int32(handleCount),
+			//Name
+			// Status
+			// UIDS
+			// GIDs
+			NumThreads: int32(pe32.CntThreads),
+			CtxSwitches: &NumCtxSwitchesStat{},
+			MemInfo: &MemoryInfoStat{
+				RSS:	pmemcounter.WorkingSetSize / 1024,
+				VMS:	pmemcounter.QuotaPagedPoolUsage / 1024,
```

Comment:
A quick glance at github.com/giampaolo/psutil shows that `VMS` should map to `pmemcounter.PagefileUsage`.

https://github.com/giampaolo/psutil/blob/0c2eb2ab9e644df8e012dd36e3d2994695bab1cc/psutil/tests/test_windows.py#L447

Reply:
https://github.com/giampaolo/psutil/blob/e0df5da398abfc0fc773736a3dfcc553d2eaa40d/psutil/_pswindows.py#L748

^ might be a better reference actually.